In [3]:
import pandas as pd
import numpy as np

df = pd.read_csv('processed_ExcelQuizMhs.csv')

df.head()   

name               P1  \
0  TIM PENGAMPU PA/TA FIK, -TUGAS AKHIR 2 - A11.4701  [192, 99, 4, 0]   
1  EGIA ROSI SUBHIYAKTO, M.Kom-SOFTWARE QUALITY A...   [43, 27, 1, 0]   
2  EGIA ROSI SUBHIYAKTO, M.Kom-REKAYASA PERANGKAT...   [27, 13, 0, 0]   
3  EGIA ROSI SUBHIYAKTO, M.Kom-SOFTWARE QUALITY A...    [27, 8, 1, 0]   
4  EGIA ROSI SUBHIYAKTO, M.Kom-REKAYASA PERANGKAT...    [31, 7, 0, 0]   

                 P2                P3               P4                P5  \
0  [182, 109, 4, 0]  [186, 106, 2, 1]  [222, 70, 3, 0]  [191, 100, 4, 0]   
1    [44, 26, 1, 0]    [45, 26, 0, 0]   [53, 17, 0, 1]    [44, 26, 0, 1]   
2    [26, 14, 0, 0]    [24, 16, 0, 0]    [35, 5, 0, 0]    [27, 13, 0, 0]   
3    [26, 10, 0, 0]    [26, 10, 0, 0]    [33, 3, 0, 0]     [27, 9, 0, 0]   
4     [32, 6, 0, 0]     [33, 5, 0, 0]    [34, 4, 0, 0]     [31, 7, 0, 0]   

                 P6                P7                P8                P9  \
0  [192, 100, 3, 0]  [184, 106, 5, 0]  [190, 100, 5, 0]  [186, 106, 3, 0]   
1    [44, 26, 0, 1]    [42, 28, 1, 0]    [45, 25, 1, 0]    [43, 28, 0, 0]   
2    [23, 17, 0, 0]    [24, 16, 0, 0]    [25, 15, 0, 0]    [24, 16, 0, 0]   
3     [29, 7, 0, 0]    [24, 12, 0, 0]     [28, 8, 0, 0]     [27, 9, 0, 0]   
4     [29, 9, 0, 0]    [28, 10, 0, 0]     [30, 8, 0, 0]     [32, 6, 0, 0]   

                P10  knowledge mastery  teaching skill  teacher skill  
0  [188, 103, 4, 0]              87.41               0          88.25  
1    [46, 25, 0, 0]              87.04               0          87.23  
2    [24, 16, 0, 0]              88.00               0          88.50  
3    [25, 11, 0, 0]              90.74               0          92.78  
4     [29, 9, 0, 0]              94.04               0          93.51

In [4]:
# remove column 'knowledge mastery'
df = df.drop(['knowledge mastery'], axis=1)
df = df.drop(['teaching skill'], axis=1)
df = df.drop(['teacher skill'], axis=1)

df.head()

name               P1  \
0  TIM PENGAMPU PA/TA FIK, -TUGAS AKHIR 2 - A11.4701  [192, 99, 4, 0]   
1  EGIA ROSI SUBHIYAKTO, M.Kom-SOFTWARE QUALITY A...   [43, 27, 1, 0]   
2  EGIA ROSI SUBHIYAKTO, M.Kom-REKAYASA PERANGKAT...   [27, 13, 0, 0]   
3  EGIA ROSI SUBHIYAKTO, M.Kom-SOFTWARE QUALITY A...    [27, 8, 1, 0]   
4  EGIA ROSI SUBHIYAKTO, M.Kom-REKAYASA PERANGKAT...    [31, 7, 0, 0]   

                 P2                P3               P4                P5  \
0  [182, 109, 4, 0]  [186, 106, 2, 1]  [222, 70, 3, 0]  [191, 100, 4, 0]   
1    [44, 26, 1, 0]    [45, 26, 0, 0]   [53, 17, 0, 1]    [44, 26, 0, 1]   
2    [26, 14, 0, 0]    [24, 16, 0, 0]    [35, 5, 0, 0]    [27, 13, 0, 0]   
3    [26, 10, 0, 0]    [26, 10, 0, 0]    [33, 3, 0, 0]     [27, 9, 0, 0]   
4     [32, 6, 0, 0]     [33, 5, 0, 0]    [34, 4, 0, 0]     [31, 7, 0, 0]   

                 P6                P7                P8                P9  \
0  [192, 100, 3, 0]  [184, 106, 5, 0]  [190, 100, 5, 0]  [186, 106, 3, 0]   
1    [44, 26, 0, 1]    [42, 28, 1, 0]    [45, 25, 1, 0]    [43, 28, 0, 0]   
2    [23, 17, 0, 0]    [24, 16, 0, 0]    [25, 15, 0, 0]    [24, 16, 0, 0]   
3     [29, 7, 0, 0]    [24, 12, 0, 0]     [28, 8, 0, 0]     [27, 9, 0, 0]   
4     [29, 9, 0, 0]    [28, 10, 0, 0]     [30, 8, 0, 0]     [32, 6, 0, 0]   

                P10  
0  [188, 103, 4, 0]  
1    [46, 25, 0, 0]  
2    [24, 16, 0, 0]  
3    [25, 11, 0, 0]  
4     [29, 9, 0, 0]

# Decision Support System - Simple Additive Weighting (SAW)

Simple Additive Weighting (SAW) is one method used to solve the problem of multi-attribute decision making. The basic concept SAW method is to find the sum of the weighted performance rating for each alternative on all attributes . SAW method requires a process of normalizing the decision matrix (X) to a scale that can be compared with all the ratings of existing alternatives.

$r_{ij} = \dfrac{x_{ij}}{Max(x_{ij})}$

$r_{ij} = \dfrac{x_{ij}}{Min(x_{ij})}$

If j is an attribute benefit then using the formula number one. If the attribute j cost then using the formula number two:

$w=\dfrac{C1}{C1 + ... + Cn}x100%$

$V_{i}=\sum_{j-1}^{n} ? 1 w_{j} r_{ij}$

The weights of all criteria are obtained by using the formula number three. With rij is the normalized performance rating of alternatives on attribute Ci Ai; i = 1,2, ..., n and j = 1,2, ..., n. Preference value alternative (vi) using the formula number four.

1. specifies criteria
2. input values the criteria
3. calculate weight criteria
4. normalization matrix
5. ranking process
6. ranking result



## 1. specifies criteria

criteria for the research is the values of student input, for example [192, 99, 4, 0]



## 2. Input values into criteria

For this we use the Likert scale from 1 to 4. 1 being the worst, 4 being the best. The values are as follows:

<table>
  <tr>
    <th>no</th>
    <th>Nilai</th>
    <th>Criteria Name</th>
    <th>Type</th>
  </tr>
  <tr>
    <td>1</td>
    <th>192</th>
    <td>Sangat baik</td>
    <td>Benefit</td>
  </tr>
  <tr>
    <td>2</td>
    <th>99</th>
    <td>Baik</td>
    <td>Benefit</td>
  </tr>
  <tr>
    <td>3</td>
    <th>4</th>
    <td>Kurang Baik</td>
    <td>Cost</td>
  </tr>
  <tr>
    <td>4</td>
    <th>0</th>
    <td>Tidak Baik</td>
    <td>Cost</td>
  </tr>

## 3. Input Alternative Value

The alternative value is the feature tied to the 2 aspects we wanted to measure. In this case, we are measuring the lecturer knowledge mastery and teaching skill.

For Knowledge Mastery aspects, we have the following values:

<table>
  <tr>
    <th>no</th>
    <th>Alternative</th>
    <th>Sangat Baik</th>
    <th>Baik</th>
    <th>Kurang Baik</th>
    <th>Tidak Baik</th>
  </tr>
  <tr>
    <th>1</th>
    <th>P1</th>
    <th>192</th>
    <th>99</th>
    <th>4</th>
    <th>0</th>
  </tr>
  <tr>
    <th>2</th>
    <th>P2</th>
    <th>182</th>
    <th>109</th>
    <th>4</th>
    <th>0</th>
  </tr>
  <tr>
    <th>3</th>
    <th>P3</th>
    <th>186</th>
    <th>106</th>
    <th>2</th>
    <th>1</th>
  </tr>
  <tr>
    <th>5</th>
    <th>P5</th>
    <th>191</th>
    <th>100</th>
    <th>4</th>
    <th>0</th>
  </tr>
  <tr>
    <th>10</th>
    <th>P10</th>
    <th>188</th>
    <th>103</th>
    <th>4</th>
    <th>0</th>
  </tr>
  

For Teaching Skill aspects, we have the following values:

<table>
  <tr>
    <th>no</th>
    <th>Alternative</th>
    <th>Sangat Baik</th>
    <th>Baik</th>
    <th>Kurang Baik</th>
    <th>Tidak Baik</th>
  </tr>
  <tr>
    <th>1</th>
    <th>P4</th>
    <th>222</th>
    <th>70</th>
    <th>3</th>
    <th>0</th>
  </tr>
  <tr>
    <th>2</th>
    <th>P6</th>
    <th>192</th>
    <th>100</th>
    <th>3</th>
    <th>0</th>
  </tr>
  <tr>
    <th>3</th>
    <th>P7</th>
    <th>184</th>
    <th>106</th>
    <th>5</th>
    <th>0</th>
  </tr>
  <tr>
    <th>5</th>
    <th>P8</th>
    <th>190</th>
    <th>100</th>
    <th>5</th>
    <th>0</th>
  </tr>
  <tr>
    <th>10</th>
    <th>P9</th>
    <th>186</th>
    <th>106</th>
    <th>3</th>
    <th>0</th>
  </tr>

### W value = [? , ? , ? , ?]

## 4. Calculate Weight Criteria

for criterion with benefit :  
$r_{ij} = \dfrac{x_{ij}}{Max(x_{ij})}$  

for criterion with cost :  
$r_{ij} = \dfrac{x_{ij}}{Min(x_{ij})}$  

\begin{matrix} r11 & r21 & r31 & r41 \\ r12 & r22 & r32 & r42 \\ r13 & r23 & r33 & r43 \\ r14 & r24 & r34 & r44 \\ r15 & r25 & r35 & r45 \end{matrix}


$r11 = \dfrac{192}{Max(192, 182, 186, 191, 188)} = 1$

$r12 = \dfrac{182}{Max(192, 182, 186, 191, 188)} = 0.94$

$r13 = \dfrac{186}{Max(192, 182, 186, 191, 188)} = 0.97$

$r14 = \dfrac{191}{Max(192, 182, 186, 191, 188)} = 0.99$

$r14 = \dfrac{188}{Max(192, 182, 186, 191, 188)} = 0.98$

\begin{matrix} 1 & r21 & r31 & r41 \\ 0.94 & r22 & r32 & r42 \\ 0.97 & r23 & r33 & r43 \\ 0.99 & r24 & r34 & r44 \\ 0.98 & r25 & r35 & r45 \end{matrix}

$r21 = \dfrac{99}{Max(99,109,106,100,103)} = 0.91$

$r22 = \dfrac{109}{Max(99,109,106,100,103)} = 1$

$r23 = \dfrac{106}{Max(99,109,106,100,103)} = 0.97$

$r24 = \dfrac{100}{Max(99,109,106,100,103)} = 0.92$

$r25 = \dfrac{103}{Max(99,109,106,100,103)} = 0.94$

\begin{matrix} 1 & 0.91 & r31 & r41 \\ 0.94 & 1 & r32 & r42 \\ 0.97 & 0.97 & r33 & r43 \\ 0.99 & 0.92 & r34 & r44 \\ 0.98 & 0.94 & r35 & r45 \end{matrix}

$r31 = \dfrac{Min(4,4,2,4,4)}{4} = 0.5$

$r32 = \dfrac{Min(4,4,2,4,4)}{4} = 1$

$r33 = \dfrac{Min(4,4,2,4,4)}{2} = 0.5$

$r34 = \dfrac{Min(4,4,2,4,4)}{4} = 1$

$r35 = \dfrac{Min(4,4,2,4,4)}{4} = 1$

\begin{matrix} 1 & 0.91 & 0.5 & r41 \\ 0.94 & 1 & 1 & r42 \\ 0.97 & 0.97 & 0.5 & r43 \\ 0.99 & 0.92 & 1 & r44 \\ 0.98 & 0.94 & 1 & r45 \end{matrix}

$r41 = \dfrac{Min(0,0,1,0,0)}{0} = 0$

$r42 = \dfrac{Min(0,0,1,0,0)}{0} = 0$

$r43 = \dfrac{Min(0,0,1,0,0)}{1} = 0$

$r44 = \dfrac{Min(0,0,1,0,0)}{0} = 0$

$r45 = \dfrac{Min(0,0,1,0,0)}{0} = 0$

\begin{matrix} 1 & 0.91 & 0.5 & 0 \\ 0.94 & 1 & 1 & 0 \\ 0.97 & 0.97 & 0.5 & 0 \\ 0.99 & 0.92 & 1 & 0 \\ 0.98 & 0.94 & 1 & 0 \end{matrix}

### Calculate Weight Preference

W value = [5,3,2,1] 
<!-- assumption -->

$V_{1} = 5*1 + 3*0.91 + 2*0.5 + 1*0 = 8.73$  
$V_{2} = 5*0.94 + 3*1 + 2*1 + 1*0 = 9.7$  
$V_{3} = 5*0.97 + 3*0.97 + 2*0.5 + 1*0 = 8.76$  
$V_{4} = 5*0.99 + 3*0.92 + 2*1 + 1*0 = 9.71$  
$V_{5} = 5*0.98 + 3*0.94 + 2*1 + 1*0 = 9.72$  